### PR TITLE
add first bits of jinja template support

### PR DESCRIPTION
**Pull request by andreaskoepf**
**[WIP]**

Notes:
- had to change the exclude configuration in `.pre-commit-config.yaml` to `exclude: "build|stubs|^bot/templates/"` because chat message templates should not be reformatted and some should not have a newline before EOF.

### CODE REVIEW

**Comment by andreaskoepf** [comment]
This PR is growing too big. Initially I only wanted to add some jinja templates .. but now it became half a bot framework. ;-)

File: bot/task_handlers.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        ...
+
+    def to_api_user(self, user: discord.User) -> protocol_schema.User:
+        return protocol_schema.User(auth_method="discord", id=user.id, display_name=user.display_name)
+
+    async def post_teaser_msg(self, template_name: str):
+        expiry_time = discord_timestamp(self.expiry_date, DiscordTimestampStyle.long_time)
+        expiry_relatve = discord_timestamp(self.expiry_date, DiscordTimestampStyle.relative_time)
+        return await self.bot.post_template(
+            template_name, task=self.task, expiry_time=expiry_time, expiry_relatve=expiry_relatve
```
Comment (yk) [inline bot/task_handlers.py:61]: typo

**Review by yk — APPROVED** [review]
looks really nice, thank you :)

File: bot/task_handlers.py (hunk 1 above)
Comment (andreaskoepf) [reply bot/task_handlers.py]: ;-) good catch, thx

**Pull request merged**